lib/stormpath-sdk/resource/collection.rb

```diff
@@ -51,6 +51,10 @@ def limit limit
     self
   end
 
+  def size
+    current_page.items.size
```

Comment:
The `size` property still needs to be removed from the `Collection` class.

I would leave the `PaginatedIterator` module the way it is because it's not the same as the `Collection` class (one represents a collection and the other is just an implementation of an iterator) and, if anything, they shouldn't even be in the same file. So, no need to change anything here for now.

Also, please update your branch which seems to be out of sync with the `development` branch.